src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/CSharpPullMemberUpCodeRefactoringProvider.cs

```diff
@@ -31,8 +31,15 @@ public CSharpPullMemberUpCodeRefactoringProvider() : base(null)
         protected override async Task<SyntaxNode> GetSelectedNodeAsync(Document document, TextSpan span, CancellationToken cancellationToken)
         {
             var refactoringHelperService = document.GetLanguageService<IRefactoringHelpersService>();
-            return await refactoringHelperService.TryGetSelectedNodeAsync(
-                document, span, n => n is MemberDeclarationSyntax || n is VariableDeclaratorSyntax, cancellationToken).ConfigureAwait(false);
+
+            var memberDecl = await refactoringHelperService.TryGetSelectedNodeAsync<MemberDeclarationSyntax>(document, span, cancellationToken).ConfigureAwait(false);
+            if (memberDecl != default)
+            {
+                return memberDecl;
+            }
+
+            var varDecl = await refactoringHelperService.TryGetSelectedNodeAsync<VariableDeclaratorSyntax>(document, span, cancellationToken).ConfigureAwait(false);
+            return varDecl;
```

Comment:
note, you can just return this.

Reply:
I know, returning awaited calls just looks weird to me for some reason.

Reply:
And you can't do `            return refactoringHelperService.TryGetSelectedNodeAsync<VariableDeclaratorSyntax>(document, span, cancellationToken);` because Task<T> isn't co?invariant. 

And since we already have private variable for memberDecl it feels ?cleaner? to have the same pattern for varDecl.

Reply:
Totally up to you.  It was just an informative comment.  :)